fcrepo4/fcrepo4
# start work centralizing config props and add fcrepo.home

**JIRA Ticket**: https://jira.lyrasis.org/browse/FCREPO-3337

# What does this Pull Request do?

Removes the following properties:

* fcrepo.ocfl.storage.root.dir
* fcrepo.ocfl.staging.dir
* fcrepo.ocfl.work.dir

Adds:

* fcrepo.home (Default: ./fcrepo)
* fcrepo.ocfl.staging (Default: ${fcrepo.home}/data/staging)
* fcrepo.ocfl.root (Default: ${fcrepo.home}/data/ocfl-root)
* fcrepo.ocfl.temp (Default: ${fcrepo.home}/data/ocfl-temp)

Repurposes `fcrepo-configs` as a module that is intended to contain all fcrepo configuration properties, and uses `fcrepo.home` to default the locations of OCFL related directories.

By default, `fcrepo.home` is configured to be the `fcrepo` directory in the current working directory. For Maven related activities, it's configured to point at `target/fcrepo-home`.

# How should this be tested?

Build everything and run `mvn jetty:run` in `fcrepo-webapp`. You should see Fedora using `target/fcrepo-home`.

Run `mvn -Dfcrepo.home=/var/tmp/fcrepo-home jetty:run`. You should see Fedora using `/var/tmp/fcrepo-home`.

Same things also work with one-click and Tomcat.

Additionally, any of the other new properties that are based on `fcrepo.home` can be overriden individually.

# Interested parties

@fcrepo4/committers
